jax/experimental/jax2tf/examples/tflite/mnist/mnist.py

```diff
@@ -0,0 +1,263 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl import app
+
+import flax  # noqa: F401
+from flax import linen as nn
+from flax import optim
+
+import jax
+from jax import random
+from jax.experimental import jax2tf
+import jax.numpy as jnp
+
+import numpy as np
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+# The code below is based on Flax Mnist Linen Example here:
+# https://github.com/google/flax/blob/master/linen_examples/mnist/mnist_lib.py
+
+
+def get_datasets():
+  """Load MNIST train and test datasets into memory."""
+  ds_builder = tfds.builder('mnist')
+  ds_builder.download_and_prepare()
+  train_ds = tfds.as_numpy(ds_builder.as_dataset(split='train', batch_size=-1))
+  test_ds = tfds.as_numpy(ds_builder.as_dataset(split='test', batch_size=-1))
+  train_ds['image'] = jnp.float32(train_ds['image']) / 255.
+  test_ds['image'] = jnp.float32(test_ds['image']) / 255.
+  return train_ds, test_ds
+
+
+class CNN(nn.Module):
```

Comment:
Can you please check how much you can reuse from examples/mnist_lib.py? It seems that you should be able to reuse the `FlaxMNIST` class from there for a lot of stuff. I think it is worth reusing because then one can focus on what is special for TFLite.

Reply:
You should be able to reuse `load_mnist` also.

Reply:
We intentionally make Flax examples *not* Python modules, so that people don't import them thus depending on the implementation details.

Reply:
Thank you! As @avital confirmed, we cannot import the examples/mnist_lib in jax2tf examples.

Reply:
I was talking about jax2tf/examples/mnist_lib.py. There is no point in replicating all the code for mnist and its training.